code/modules/shield_generators/shield_generator.dm

```diff
@@ -48,22 +51,36 @@
 
 
 /obj/machinery/power/shield_generator/Initialize()
-	. = ..()
+	..()
 	connect_to_network()
-
 	mode_list = list()
 	for(var/st in subtypesof(/datum/shield_mode/))
 		var/datum/shield_mode/SM = new st()
 		mode_list.Add(SM)
+	GLOB.moved_event.register(src, src, .proc/update_overmap_shield_list)
+	. = INITIALIZE_HINT_LATELOAD
 
+/obj/machinery/power/shield_generator/LateInitialize()
+	. = ..()
+	update_overmap_shield_list()
 
 /obj/machinery/power/shield_generator/Destroy()
 	shutdown_field()
 	field_segments = null
 	damaged_segments = null
 	mode_list = null
+	GLOB.moved_event.unregister(src, src)
 	. = ..()
```

Comment:
should probably also remove self from overmap object's list

Reply:
Clears it in the update proc.